### PR TITLE
test(dtslint): add dtslint test on count operator (#4093)

### PR DESCRIPTION
**Pull request by dkosasih**
**Description:** Add count dtslint test

**Related issue (if exists):** #4093

### CODE REVIEW

File: spec-dtslint/operators/count-spec.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { of } from 'rxjs';
+import { count } from 'rxjs/operators';
+
+it('should always infer number', () => {
+  const o = of(1, 2, 3).pipe(count(x => x > 1)); // $ExpectType Observable<number>
```
Comment (timdeschryver) [inline spec-dtslint/operators/count-spec.ts]: I believe we should also test the predicate overloads.

```ts
predicate: (value: T, index: number, source: Observable<T>)
```

File: spec-dtslint/operators/count-spec.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+});
+
+it('should accept empty parameter', () => {
+  const o = of(1, 2, 3).pipe(count()); // $ExpectType Observable<number>
+});
+
+it('should expect function parameter', () => {
+  const o = of(1, 2, 3).pipe(count(9)); // $ExpectError
```
Comment (timdeschryver) [inline spec-dtslint/operators/count-spec.ts]: We should also test if the value is the same type as the source.
For example 

```ts
const o = of(1, 2, 3).pipe(count((value: string) => value !== ''); // $ExpectError
```

File: spec-dtslint/operators/count-spec.ts (hunk 1 above)
Comment (dkosasih) [reply spec-dtslint/operators/count-spec.ts]: Added few more tests around the predicate overloads types. 
Please let me know if they are of a good value. 
Thanks

File: spec-dtslint/operators/count-spec.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { of } from 'rxjs';
+import { count, buffer } from 'rxjs/operators';
+
+it('should always infer number', () => {
+  const o = of(1, 2, 3).pipe(count(x => x > 1)); // $ExpectType Observable<number>
```
Comment (cartant) [inline spec-dtslint/operators/count-spec.ts]: With `count`, the return type of the piped observable will always be `Observable<number>`, so I think that for the tests, the source observable type should be something other than `number` - to make sure that the output type comes from the `count` operator and not the source. E.g.:

```ts
const o = of('1', '2', '3').pipe(count(x => x !== '1')); // $ExpectType Observable<number>
```

Comment (dkosasih) [reply spec-dtslint/operators/count-spec.ts]: @cartant I have added another row to both number or string will always return number

**Pull request merged**